src/python/pants/backend/go/util_rules/third_party_pkg.py

```diff
@@ -229,6 +229,10 @@ async def analyze_module_dependencies(request: ModuleDescriptorsRequest) -> Modu
         if "Main" in mod_json and mod_json["Main"]:
             continue
 
+        # Skip first-party modules referenced from other first-party modules.
```

Comment:
Should we add a TODO here about using the replacement value for dep inference? Is that a thing we will want to do, or is this dep inferrable some other way already?

Reply:
It is not yet able to infer through the replacement value, but that would be a great addition in the future! I'll add a TODO. For now, I've been using explicit references between packages that cross the module boundaries, but automatic would be much cleaner.

```suggestion
        # Skip first-party modules referenced from other first-party modules.
        # TODO These cross-module references could be used for dependency inference
```